lib/retriable.rb

```diff
@@ -59,13 +62,23 @@ def retriable(opts = {})
       begin
         return Timeout.timeout(timeout) { return yield(try) } if timeout
         return yield(try)
-      rescue *[*exception_list] => exception
+      rescue *[*rescue_list] => exception
         if on.is_a?(Hash)
           raise unless exception_list.any? do |e|
             exception.is_a?(e) && ([*on[e]].empty? || [*on[e]].any? { |pattern| exception.message =~ pattern })
           end
         end
 
+        if ignore.is_a?(Hash)
+          raise if ignore_list.any? do |e|
```

Comment:
What should happen if the user puts the same exception in both lists? In this implementation, `ignore` will override `on` right?